packages/fxa-auth-server/lib/error.js

```diff
@@ -1258,6 +1258,15 @@ AppError.subscriptionAlreadyExists = () => {
   });
 };
 
+AppError.userAlreadySubscribedToProduct = () => {
+  return new AppError({
+    code: 409,
+    error: 'Already subscribed to product with different plan',
+    errno: ERRNO.SUBSCRIPTION_ALREADY_EXISTS,
```

Comment:
This might be good for a follow up issue. It's slipped my mind at the moment, but there was another spot (maybe in subscription upgrades?) where I was getting an error number that was the same error number in a handful of error cases, and having distinct errors would have helped narrow down the problem faster.

Reply:
@jodyheavener I don't think in this case it would narrow things down for us much at all -  since a Plan is just a child of a Product. There is additional data in the message that indicates that rather than being actively subscribed to the Plan, they are subscribed to the product with a different plan. In both cases, should we find the error was given incorrectly, we would find ourselves in the same situation: looking up a Customer record in Stripe and verifying their active Subscriptions. Ultimately, the reason for failure is the same for both: The user already has access to the product, and we don't want them to accidentally purchase what they already own.

Reply:
This is how the information for subscriptions is displayed in Stripe, so you can see that it wouldn't affect debugging as to a failure because of Plan vs. Product:
<img width="272" alt="Screen Shot 2020-04-01 at 1 22 12 PM" src="https://user-images.githubusercontent.com/3428280/78182881-db46a300-741b-11ea-9405-9d38188c6433.png">